File: old_versions/ProgAndPlay/Client_Interfaces/java/exemple/Mission7.java
package exemple;

import exemple.ConstantList_KP_4_1;

import pp.PP;
import pp.Unit;

public class Mission7 {
	public static void main (String[] args){
		try {
			// creer une interface avec le jeu
			PP pp = new PP();
			// ouverture du jeu
			pp.open();
			// recherche de l'assembleur
			int i = 0;
			Unit u = pp.getUnitAt(PP.Coalition.MY_COALITION, i);
			while (i < pp.getNumUnits(PP.Coalition.MY_COALITION) && u.getType() != ConstantList_KP_4_1.ASSEMBLER){
				// passer a l'unite suivante
				i = i + 1;
				u = pp.getUnitAt(PP.Coalition.MY_COALITION, i);
			}
			if (u.getType() == ConstantList_KP_4_1.ASSEMBLER){
				Unit assembleur = u;
				// reparation des unites
				for (i = 0 ; i < pp.getNumUnits(PP.Coalition.MY_COALITION) ; i++){
					u = pp.getUnitAt(PP.Coalition.MY_COALITION, i);
					if (u.getHealth() < u.getMaxHealth()){
						assembleur.command(ConstantList_KP_4_1.REPAIR, u);
						while (u.getHealth() < u.getMaxHealth()){
							System.out.println("Sante de l'unite "+u.getId()+" : "+u.getHealth()+"/"+u.getMaxHealth());
						}
					}
				}
			}
			// fermer le jeu
			pp.close();
		} catch (Exception e){
			System.out.println (e.getMessage());
		}
	}
}

File: old_versions/ProgAndPlay/Client_Interfaces/python/mission7.py
from constantList_KP4_1 import *
from pp import *

# Ouverture de la connexion avec le jeu
PP_Open()
# Recherche de l'assembleur
i = 0
u = PP_GetUnitAt(MY_COALITION, i)
while i < PP_GetNumUnits(MY_COALITION) and PP_Unit_GetType(u) != ASSEMBLER:
    # Passer à l'unité suivante
    i += 1
    u = PP_GetUnitAt(MY_COALITION, i)
    
if PP_Unit_GetType(u) == ASSEMBLER:
    assembleur = u
    # Réparation des unités
    i = 0
    while i < PP_GetNumUnits(MY_COALITION):
        u = PP_GetUnitAt(MY_COALITION, i)
        if PP_Unit_GetHealth(u) < PP_Unit_GetMaxHealth(u):
            PP_Unit_ActionOnUnit(assembleur, REPAIR, u)
            while PP_Unit_GetHealth(u) < PP_Unit_GetMaxHealth(u):
                # Afficher la santée courante
                print("Sante de l'unite", u, ":", PP_Unit_GetHealth(u), "/", PP_Unit_GetMaxHealth(u))
        # Passer à l'unité suivante
        i += 1
# Fermer la connexion avec le jeu
PP_Close()
